packages/web3-providers-ws/src/types.ts

```diff
@@ -0,0 +1,41 @@
+import { OutgoingHttpHeaders } from 'http';
+import { IClientConfig } from 'websocket';
+
+export type ReconnectOptions = {
+    auto: boolean;
+    delay: number;
+    maxAttempts: number;
+    onTimeout: boolean;
+};
+export interface WebSocketOptions {
+    protocol?: string;
+    headers?: OutgoingHttpHeaders;
+    requestOptions?: object;
+    clientConfig?: IClientConfig;
+    customTimeout?: number;
+    reconnectOptions?: ReconnectOptions;
+    timeout?: number;
+    reconnectDelay?: number;
+    origin?: string;
+}
+
+// TODO: Fancy errors
+export enum WSErrors {
+    ConnectionTimeout = 'Connection Timeout ',
+    ConnectionNotOpenError = 'Connection Not Open Error ',
+    PendingRequestsOnReconnectingError = 'Pending Requests On Reconnecting Error ',
+    MaxAttemptsReachedOnReconnectingError = 'Max Attempts Reached On Reconnecting Error ',
+    InvalidConnection = 'Invalid Connection ',
+}
+
+export enum WSStatus {
+    Error = 'Error ',
+    Reconnect = 'Reconnect ',
+}
+
+export interface JsonRpcResponse {
```

Comment:
Can we use [RpcResponse](https://github.com/ChainSafe/web3.js/blob/4.x/packages/web3-core-types/src/types.ts#L169) here? Looks like we need to update `result` to optional and add `error`

Reply:
I left some types intentionally to be changed in next PR. [like](https://github.com/ChainSafe/web3.js/blob/ed547ae2ab654dab7269ccd57b3fe6ee7f3edc2e/packages/web3-providers-ws/src/index.ts#L305) , (specially in `parseResponse` and `onMessage`)

Reply:
Ok, I've created an issue to track this #4255